ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeSignatureValidator.java

```diff
@@ -0,0 +1,194 @@
+import tech.pegasys.teku.spec.datastructures.operations.versions.altair.ValidateableSyncCommitteeSignature;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.altair.BeaconStateAltair;
+import tech.pegasys.teku.spec.logic.common.util.SyncCommitteeUtil;
+import tech.pegasys.teku.statetransition.validation.InternalValidationResult;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class SyncCommitteeSignatureValidator {
+  private static final Logger LOG = LogManager.getLogger();
+  private final Set<UniquenessKey> seenIndices =
+      LimitedSet.create(VALID_SYNC_COMMITTEE_SIGNATURE_SET_SIZE);
+  private final Spec spec;
+  private final RecentChainData recentChainData;
+  private final SyncCommitteeStateUtils syncCommitteeStateUtils;
+
+  public SyncCommitteeSignatureValidator(
+      final Spec spec,
+      final RecentChainData recentChainData,
+      final SyncCommitteeStateUtils syncCommitteeStateUtils) {
+    this.spec = spec;
+    this.recentChainData = recentChainData;
+    this.syncCommitteeStateUtils = syncCommitteeStateUtils;
+  }
+
+  public SafeFuture<InternalValidationResult> validate(
+      final ValidateableSyncCommitteeSignature validateableSignature) {
+
+    final SyncCommitteeSignature signature = validateableSignature.getSignature();
+
+    final Optional<SyncCommitteeUtil> maybeSyncCommitteeUtil =
+        spec.getSyncCommitteeUtil(signature.getSlot());
+    if (maybeSyncCommitteeUtil.isEmpty()) {
+      LOG.trace(
+          "Rejecting sync committee signature because the fork active at slot {} does not support sync committees",
+          signature.getSlot());
+      return SafeFuture.completedFuture(REJECT);
+    }
+    final SyncCommitteeUtil syncCommitteeUtil = maybeSyncCommitteeUtil.get();
+
+    // [IGNORE] The signature's slot is for the current slot, i.e. sync_committee_signature.slot ==
+    // current_slot.
+    if (recentChainData.getCurrentSlot().isEmpty()
+        || !signature.getSlot().equals(recentChainData.getCurrentSlot().orElseThrow())) {
+      LOG.trace("Ignoring sync committee signature because it is not from the current slot");
+      return SafeFuture.completedFuture(IGNORE);
+    }
+
+    // [IGNORE] There has been no other valid sync committee signature for the declared slot for the
+    // validator referenced by sync_committee_signature.validator_index.
+    final UniquenessKey uniquenessKey = getUniquenessKey(signature);
+    if (seenIndices.contains(uniquenessKey)) {
+      return SafeFuture.completedFuture(IGNORE);
+    }
+
+    // [IGNORE] The block being signed over (sync_committee_signature.beacon_block_root) has been
+    // seen (via both gossip and non-gossip sources).
+    if (!recentChainData.containsBlock(signature.getBeaconBlockRoot())) {
+      LOG.trace("Ignoring sync committee signature because beacon block is not known");
+      return SafeFuture.completedFuture(IGNORE);
```

Comment:
Should we be saving this?:
```suggestion
      return SafeFuture.completedFuture(SAVE_FOR_FUTURE);
```

Reply:
There's no provision in the network spec to save these for the future.  I'm not sure there's much value in adding the complexity of queuing given they're only valid for one slot - if the block is late all the votes will likely be for the previous block anyway and will be useless by the time we get the next block.